core/src/main/java/org/elasticsearch/env/Environment.java

```diff
@@ -108,28 +109,32 @@
     }
 
     public Environment(Settings settings) {
+        this(PathUtils.getDefaultFileSystem(), settings);
+    }
+
+    public Environment(FileSystem fs, Settings settings) {
```

Comment:
See the diff in my comment above; I already removed the change to `Environment` in that diff but I'm inquiring about your thoughts on the hack around `java.io.tmdir`.